lib/android-tools-info.ts

```diff
@@ -0,0 +1,318 @@
+				errors.push({
+					warning: `Javac version ${installedJavaVersion} is not supported. You have to install at least ${AndroidToolsInfo.MIN_JAVA_VERSION}.`,
+					additionalInformation: additionalMessage,
+					platforms: [Constants.ANDROID_PLATFORM_NAME]
+				});
+			}
+		} else {
+			errors.push({
+				warning: "Error executing command 'javac'. Make sure you have installed The Java Development Kit (JDK) and set JAVA_HOME environment variable.",
+				additionalInformation: additionalMessage,
+				platforms: [Constants.ANDROID_PLATFORM_NAME]
+			});
+		}
+
+		return errors;
+	}
+
+	public async getPathToAdbFromAndroidHome(): Promise<string> {
+		if (this.androidHome) {
+			let pathToAdb = path.join(this.androidHome, "platform-tools", "adb");
+			try {
+				await this.childProcess.execFile(pathToAdb, ["help"]);
+				return pathToAdb;
+			} catch (err) {
+				return null;
+			}
+		}
+
+		return null;
+	}
+
+	public validateAndroidHomeEnvVariable(): NativeScriptDoctor.IWarning[] {
+		const errors: NativeScriptDoctor.IWarning[] = [];
+		const expectedDirectoriesInAndroidHome = ["build-tools", "tools", "platform-tools", "extras"];
+
+		if (!this.androidHome || !this.fs.exists(this.androidHome)) {
+			errors.push({
+				warning: "The ANDROID_HOME environment variable is not set or it points to a non-existent directory. You will not be able to perform any build-related operations for Android.",
+				additionalInformation: "To be able to perform Android build-related operations, set the `ANDROID_HOME` variable to point to the root of your Android SDK installation directory.",
+				platforms: [Constants.ANDROID_PLATFORM_NAME]
+			});
+		} else if (expectedDirectoriesInAndroidHome.map(dir => this.fs.exists(path.join(this.androidHome, dir))).length === 0) {
+			errors.push({
+				warning: "The ANDROID_HOME environment variable points to incorrect directory. You will not be able to perform any build-related operations for Android.",
+				additionalInformation: "To be able to perform Android build-related operations, set the `ANDROID_HOME` variable to point to the root of your Android SDK installation directory, " +
+				"where you will find `tools` and `platform-tools` directories.",
+				platforms: [Constants.ANDROID_PLATFORM_NAME]
+			});
+		}
+
+		return errors;
+	}
+
+	public getPathToEmulatorExecutable(): string {
+		if (!this.pathToEmulatorExecutable) {
+			const emulatorExecutableName = "emulator";
+
+			this.pathToEmulatorExecutable = emulatorExecutableName;
+
+			if (this.androidHome) {
+				// Check https://developer.android.com/studio/releases/sdk-tools.html (25.3.0)
+				// Since this version of SDK tools, the emulator is a separate package.
+				// However the emulator executable still exists in the "tools" dir.
+				const pathToEmulatorFromAndroidStudio = path.join(this.androidHome, emulatorExecutableName, emulatorExecutableName);
+
+				if (this.fs.exists(pathToEmulatorFromAndroidStudio)) {
```

Comment:
this check will not work on Windows as the executable has `.exe` in the name. You can check the latest implementation [here](https://github.com/telerik/mobile-cli-lib/pull/907/files#diff-ed599f397679aa4cd0c786eae0767a7dR60)